ast.go

```diff
@@ -111,6 +111,29 @@ func (d DataType) LessThan(other DataType) bool {
 	return other != Unknown && other < d
 }
 
+// Zero returns the zero value for the DataType.
+// The return value of this method, when sent back to InspectDataType,
+// may not produce the same value.
+func (d DataType) Zero() interface{} {
+	switch d {
```

Comment:
If you store the zero values in unexported, package-level variables, you'll avoid the `interface{}` allocation every time it is called:

```go
var (
    // must assign to interface{}
    zeroFloat64 interface{} = float64(0)
    // ...
)

func (d DataType) Zero() interface{} {
    switch d {
    case Float:
        return zeroFloat64
    }
}
```

Reply:
Done.